prime-angular-frontend/src/app/modules/site-registration/pages/device-provider-page/device-provider-page-form-state.class.ts

```diff
@@ -0,0 +1,73 @@
+import { FormArray, FormBuilder, FormGroup, Validators } from '@angular/forms';
+
+import { AbstractFormState } from '@lib/classes/abstract-form-state.class';
+import { FormControlValidators } from '@lib/validators/form-control.validators';
+
+import { IndividualDeviceProvider } from '@registration/shared/models/individual-device-provider.model';
+
+export class DeviceProviderPageFormState extends AbstractFormState<IndividualDeviceProvider[]> {
+  public constructor(
+    private fb: FormBuilder
+  ) {
+    super();
+
+    this.buildForm();
+  }
+
+  public get individualDeviceProviders(): FormArray {
+    return this.formInstance.get('individualDeviceProviders') as FormArray;
+  }
+
+  public get json(): IndividualDeviceProvider[] {
+    if (!this.formInstance) {
+      return;
+    }
+
+    return this.individualDeviceProviders.getRawValue();
+  }
+
+  public patchValue(providers: IndividualDeviceProvider[]): void {
+    if (!this.formInstance || !Array.isArray(providers)) {
+      return;
+    }
+
+    this.individualDeviceProviders.clear(); // Clear out existing indices
+
+    providers.forEach((p: IndividualDeviceProvider) => this.push(p));
```

Comment:
The method name `push` is slightly deceptive since you're creating a group, patching, and then pushing.  Also, it looks like you only use this once so you could drop the method and inline its content within this `patchValue`